document/RuleMetadataProfile.mdk

```diff
@@ -1 +1,165 @@
-# Rule Metadata Profile { #rule-metadata-profile }
\ No newline at end of file
+# Rule Metadata Profile { #rule-metadata-profile }
+
+## General
+
+It is sometimes useful for an issue management system to provide information to the end user
+about the rules that were violated by the issues in a log file.
+For example, the user might want to know whether the issue is a potential security risk,
+or where to find additional documentation for the issue.
+We refer to this information as rule metadata.
+
+It is possible for an issue management system to have built-in knowledge of
+the metadata for the rules defined by a set of well-known static analysis tools.
+It is also possible for the system to provide a means for end users to upload
+metadata for rules defined by specialized tools used in their organizations.
+But some tools enable end users to define their own rules by means of a plug-in
+mechanism.
+In this case, it is useful if the metadata for these rules can be included
+in the log file itself.
+
+The Rule Metadata Profile defines additional optional elements of the SARIF format
+that are intended for use in this scenario.
+
+## `runLog` object { #rule-metadata-profile-runLog }
+
+### General { #rule-metadata-profile-runLog-general }
+
+The Rule Metadata Profile defines one additional property in the `runLog` object (&sect;[#runLog]).
+
+### `ruleInfo` property { #ruleInfo }
+
+A `runLog` object (&sect;[#runLog]) may contain a property named `ruleInfo`
+whose value is an array of `ruleDescriptor` (&sect;[#ruleDescriptor]) objects.
+
+## `ruleDescriptor` object { #ruleDescriptor }
+
+### General { #ruleDescriptor-General }
+
+A `ruleDescriptor` object describes a discrete, potentially configurable,
+unit of analysis as implemented by an individual tool.
+
+### `id` property { #ruleDescriptor-id }
+
+A `ruleDescriptor` object shall contain a property named `id` whose value is a string containing
+a stable, opaque identifier for the rule.
+
+~ IsoExample
+`"CA2101"`
+~
+
+### `name` property { #ruleDescriptor-name }
+
+A `ruleDescriptor` object may contain a property named `name` whose value is a string containing
+a readable rule identifier. A rule name is suitable for use as a key to other data or to
+otherwise identify a rule (e.g., within an in-source suppression) but is not guaranteed to
```

Comment:
1. It's relatively stable. :) How about this 'A rule name is suitable in contexts where a readable identifier is preferable and where the non-guarantee of stability is irrelevant or not a significant concern.

[back to the comment, in practice a rule name will only change occasionally, typically in a version update of a tool. some tools couple a readable name with an opaque identifier to get the best of both worlds. CA1001.WatchOutForThisSpecificCondition. In the previous example, the opaque id on the left is parsed for any 'stable key' needs. The readable name suffix is there simply to support developers looking at the information, to understand what class of issue CA1001 refers to]